Jakarta - EE 10 - @Inject should work instead of the deprecated @Context in RESTEasy Classic

### Describe the bug

In JAX-RS we used to use `@Context` to inject several managed objects, including `UriInfo`:

```
import jakarta.ws.rs.core.Context;

@Path("/hello")
public class GreetingResource {

    @Context
    UriInfo uriInfo;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        System.out.println("Base URI: " + uriInfo.getBaseUri());
        return "Hello RESTEasy";
    }
}
```

With Jakarta EE 10 and JAX-RS 4 we can use `@Inject` [instead](https://jakarta.ee/specifications/restful-ws/4.0/):

```    
@Inject
UriInfo uriInfo;
```

ATM, this code breaks with RESTEasy Classic (it works with RESTEasy Reactive) with the following exception:

```
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type jakarta.ws.rs.core.UriInfo and qualifiers [@Default]
        - java member: org.acme.GreetingResource#uriInfo
        - declared on CLASS bean [types=[org.acme.GreetingResource, java.lang.Object], qualifiers=[@Default, @Any], target=org.acme.GreeetingResource]
````

```
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy-reactive</artifactId>
    </dependency>
```

See https://jakarta.ee/specifications/restful-ws/4.0
Relates to: https://github.com/jakartaee/rest/issues/951

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.0.Alpha1

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -6191,6 +6191,7 @@
                                 <recipe>io.quarkus.smallrye</recipe>
                                 <recipe>io.quarkus.bom.resteasy-microprofile</recipe>
                                 <recipe>io.quarkus.bom.resteasy-spring-web</recipe>
+                                <recipe>io.quarkus.bom.resteasy-cdi</recipe>
                                 <recipe>io.quarkus.jakarta-json</recipe>
                                 <recipe>io.quarkus.keycloak-admin-client</recipe>
                                 <recipe>io.quarkus.infinispan</recipe>

--- a/jakarta/rewrite.yml
+++ b/jakarta/rewrite.yml
@@ -780,12 +780,24 @@ recipeList:
       addIfMissing: true
 ---
 type: specs.openrewrite.org/v1beta/recipe
+name: io.quarkus.bom.resteasy-cdi
+displayName: Add RESTEasy CDI to BOM
+recipeList:
+  - org.openrewrite.maven.AddManagedDependency:
+      groupId: org.jboss.resteasy
+      artifactId: resteasy-cdi
+      version: "${resteasy.version}"
+---
+type: specs.openrewrite.org/v1beta/recipe
 name: io.quarkus.resteasy-common
-displayName: Add RESTEasy MicroProfile Config to RESTEasy Common
+displayName: Add RESTEasy CDI and MicroProfile Config to RESTEasy Common
 recipeList:
   - org.openrewrite.maven.AddDependencyNoQuestionsAsked:
       groupId: org.jboss.resteasy.microprofile
       artifactId: microprofile-config
+  - org.openrewrite.maven.AddDependencyNoQuestionsAsked:
+      groupId: org.jboss.resteasy
+      artifactId: resteasy-cdi
   - org.openrewrite.maven.ExcludeDependency:
       groupId: org.jboss.spec.javax.ws.rs
       artifactId: jboss-jaxrs-api_3.0_spec

